### PR TITLE
iox-#1032 replace 'IOX_EXPECTS'/'IOX_ENSURES' with 'IOX_ENFORCE'

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Add a second reviewer for complex new features or larger refactorings
1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. ~[ ] Changelog updated [in the unreleased section][changelog] including API breaking changes~
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

This PR replaces the legacy error handler macros ('IOX_ENFORCE' and 'IOX_EXPECTS') with the new `IOX_ENFORCE` macro and adjusts the tests. Additionally, the error output is improved.

The changelog will be updated with the next PR which also removes all the TODOs and updates the error handling md files.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Relates to #1032
- Relates to #2011

### CODE REVIEW

**Comment by elBoberido** [comment]
@FerdinandSpitzschnueffler @mossmaurice @dkroenke @MatthiasKillat @budrus @elfenpiff feel free to add yourself as reviewer

File: iceoryx_hoofs/posix/utility/source/posix_scheduler.cpp [hunk 1]
```diff
@@ -47,16 +39,8 @@ int32_t getSchedulerPriorityMaximum(const Scheduler scheduler) noexcept
...
-                "The \"sched_get_priority_max\" should never fail. This can only be caused by an internal logic "
-                "error or a non posix compliant system.");
-
-        // NOLINTJUSTIFICATION Required to provide an error message to the user
-        // NOLINTNEXTLINE(hicpp-no-array-decay,cppcoreguidelines-pro-bounds-array-to-pointer-decay)
-        IOX_ENSURES(false
-                    && "This should never happen! Either the system is not posix compliant or and invalid integer was "
-                       "casted to the enum class Scheduler.");
-        return -1;
+        IOX_PANIC("The \"sched_get_priority_min\" should never fail! Either the system is not POSIX compliant or and "
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_hoofs/posix/utility/source/posix_scheduler.cpp:42]:
```suggestion
        IOX_PANIC("The \"sched_get_priority_max\" should never fail! Either the system is not POSIX compliant or an "
```

Comment (FerdinandSpitzschnueffler) [reply iceoryx_hoofs/posix/utility/source/posix_scheduler.cpp]: This used call is actually `sched_get_priority_max`.

Comment (elBoberido) [reply iceoryx_hoofs/posix/utility/source/posix_scheduler.cpp]: Just wanted to see if you pay attention ;)

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Pull request merged**